packages/firebase_core/firebase_core_web/lib/src/firebase_js.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@JS('firebase')
+library firebase_js;
+
+import 'package:js/js.dart';
+
+@JS()
+class App {
+  external String get name;
+  external Options get options;
+}
+
+@JS()
+@anonymous
+class Options {
+  external factory Options({
+    String apiKey,
+    String authDomain,
+    String databaseURL,
+    String projectId,
+    String storageBucket,
+    String messagingSenderId,
+    String appId,
+    String measurementId,
+  });
+
+  external String get apiKey;
+  external String get authDomain;
+  external String get databaseURL;
+  external String get projectId;
+  external String get storageBucket;
+  external String get messagingSenderId;
+  external String get appId;
+  external String get measurementId;
+}
+
+@JS()
+external List<App> get apps;
+
+@JS()
+external App initializeApp(Options options, String name);
+
+@JS()
+external App app(String name);
```

Comment:
The package:js only way of doing this:

```dart
@JS()
library firebase_js;

// App, Options...

@JS()
class FirebaseCore {
  external List<App> get apps;
  external App initializeApp(Options options, String name);
  external App app(String name);
}

@JS("firebase")
external FirebaseCore get firebase;
```

Then on the frontend code you can do:

```dart
import 'src/firebase_js.dart'; // No need for 'as firebase', because everything is below the getter

if (firebase != null) {
  // Firebase not loaded!
}
```

That way you don't have to mix `dart:js` syntax with `package:js` syntax.

Reply:
Thanks! Done.